internal/extension/oop.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2025 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package extension
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"syscall"
+	"time"
+
+	extpb "github.com/kuadrant/kuadrant-operator/pkg/extension/grpc/v0"
+
+	"github.com/go-logr/logr"
+	"google.golang.org/grpc"
+)
+
+const defaultUnixSocket = ".grpc.sock"
+
+type OOPExtension struct {
+	name       string
+	executable string
```

Comment:
If we would like to send arguments to the `exec.Cmd` we could change the signature to maybe a `[]string`, being `executable[0]` the command and the rest the args.

Reply:
This proposes another approach, as this actually does pass arguments, well _one_ argument. That's the `socket`. So instead of "just" passing "any arg", the API here specifies the semantic of the arg. And actually how the value is passed to the child is abstracted from the API. As it turns out today the socket is passed (see above as to why) via an argument, but it could use any other mean to achieve the same goal, the caller of this API should _not_ know about how that's achieved